skietbaan_fe/src/components/ScoreCapture.js

```diff
@@ -1,11 +1,141 @@
 
 export default class search extends Component {
-    render() {
-        return (
-            <div>
+  constructor(props) {
+    super(props);
+    this.state = {
+      currState: 1,
+      clicked: null,
+      score: 0,
+      hideCamera: false,
+      imageContent: "",
+      competitionName: "",
+      validForm: false,
+    }
+    this.CompetitionClicked = this.CompetitionClicked.bind(this);
+    this.handleScore = this.handleScore.bind(this);
+    this.TakePhoto = this.TakePhoto.bind(this);
+    this._base64ToArrayBuffer = this._base64ToArrayBuffer.bind(this);
+    this.SubmitScore = this.SubmitScore.bind(this);
 
-            </div>
-                )
+  }
+  handleScore({ target }) {
+    this.setState({
+      [target.name]: target.value,
+    });
+    let isValid = true;
+    let stateUpdate = {
+      invalidScore: false,
+    }
+    if (!validateScore(this.state.score)) {
+      stateUpdate.invalidPassword= true;
+      isValid = false;
+    };   
+    this.setState({
+      ...stateUpdate ,
+      validForm: isValid
+    }); 
+  }
+  CompetitionClicked(item, competitionName) {
+    this.setState({
+      currState: 2,
+      clicked: item,
+      competitionName: competitionName
+    });
+  }
+  _base64ToArrayBuffer(base64) {
+    var binary_string = base64;
+    var len = binary_string.length;
+    var bytes = new Uint8Array(len);
+    for (var i = 0; i < len; i++) {
+      bytes[i] = binary_string.charCodeAt(i);
+    }
+    return bytes.buffer;
+  }
+  CameraClicked() {
+    this.setState({
+      currState: 3
+    });
+    let video = document.getElementById('video');
+
+    if (navigator.mediaDevices && navigator.mediaDevices.getUserMedia) {
+      navigator.mediaDevices.getUserMedia({ video: true }).then(function (stream) {
+        video.srcObject = stream;
+        video.play();
+      });
+
+    };
+  }
+  SubmitScore() {
+   if(this.state.validForm){
+    let RequestObject = {
+      "UserScore": this.state.score,
+      "PictureURL": this.state.imageContent,
+      "Competition": this.state.competitionName
+    }
+    fetch("https://api.skietbaan.retrotest.co.za/api/Scores", {
+        method: 'post',
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(RequestObject)
+      }).then(function(response) {
+        return response.json();
+      }).then( function(data) {
+      }).catch(function(data) {
+      });
+    }
+  }
+
+  TakePhoto() {
+    let canvas = document.getElementById('canvas');
+    let context = canvas.getContext('2d');
+    let video = document.getElementById('video');
+    context.drawImage(video, 0, 0, 640, 480);
+    const image = new Image()
+    image.src = canvas.toDataURL();
+    console.log("photo")
+    this.setState({
+      currState: 4,
+      hideCamera: true,
+      imageContent: image.src
+    });
+  }
+  render() {
+    let competitionItem = [];
+    for (let i = 0; i < 6; i++) {
```

Comment:
What is 6?

Reply:
Because the controller for displaying competition names isn't completed yet. I am using this as a way to select a competition 0 to 5 so that it can take me to the next part of the page. So I can add a score to that until the controller is completed